### PR TITLE
Added documentation for spectral layout (#2971)

### PR DESCRIPTION
**Pull request by j-petit**
Added documentation as proposed in #2971

### CODE REVIEW

File: networkx/drawing/layout.py [hunk 1]
```diff
@@ -736,6 +736,11 @@ def _kamada_kawai_costfn(pos_vec, np, invdist, meanweight, dim):
 def spectral_layout(G, weight='weight', scale=1, center=None, dim=2):
     """Position nodes using the eigenvectors of the graph Laplacian.
 
+    Using the unnormalized Laplacion, the layout shows possible clusters of
+    nodes which are an approximation of the ratio cut. If n is the number of
+    dimensions then the positions are the entries of the n eigenvectors
```
Comment (dschult) [inline networkx/drawing/layout.py]: Can you replace ```n``` with ```dim```?

File: networkx/drawing/nx_pylab.py [hunk 2]
```diff
@@ -1007,7 +1007,12 @@ def draw_random(G, **kwargs):
 
 
 def draw_spectral(G, **kwargs):
-    """Draw the graph G with a spectral layout.
+    """Draw the graph G with a spectral 2D layout.
+
+    Using the unnormalized Laplacion, the layout shows possible clusters of
+    nodes which are an approximation of the ratio cut. The positions are the
+    entries of the second and third eigenvectors corresponding to the
+    eigenvalues in ascending order.
```
Comment (dschult) [inline networkx/drawing/nx_pylab.py]: maybe add:  starting from the second eigenvalue.

**Comment by j-petit** [comment]
Thanks for the feedback. Should I combine the commits into a single one?

**Comment by dschult** [comment]
No need to "squash" the commits.  github does that by default for each PR when we merge it.
Thanks!

**Pull request merged**